src/visor/blockchain_parser.go

```diff
@@ -93,35 +82,24 @@ func (bcp *BlockchainParser) Stop() {
 }
 
 func (bcp *BlockchainParser) parseTo(bcHeight uint64) error {
-	if bcp.parsedHeight == 0 {
-		// logger.Critical("historydb parse %d/%d", bcp.parsedHeight, bcHeight)
-		for i := uint64(0); i <= bcHeight-bcp.parsedHeight; i++ {
-			b := bcp.bc.GetBlockInDepth(bcp.parsedHeight + i)
-			if b == nil {
-				return fmt.Errorf("no block exist in depth:%d", bcp.parsedHeight+i)
-			}
+	parsedHeight := bcp.historyDB.ParsedHeight()
 
-			if err := bcp.historyDB.ProcessBlock(b); err != nil {
-				return err
-			}
+	for i := int64(0); i < int64(bcHeight)-parsedHeight; i++ {
+		b := bcp.bc.GetBlockInDepth(uint64(parsedHeight + i + 1))
+		if b == nil {
+			return fmt.Errorf("no block exist in depth:%d", parsedHeight+i+1)
 		}
-		// logger.Critical("historydb parse %d/%d", bcHeight, bcHeight)
-	} else {
-		// logger.Critical("historydb parse %d/%d", bcp.parsedHeight, bcHeight)
-		for i := uint64(0); i < bcHeight-bcp.parsedHeight; i++ {
-			b := bcp.bc.GetBlockInDepth(bcp.parsedHeight + i + 1)
-			if b == nil {
-				return fmt.Errorf("no block exist in depth:%d", bcp.parsedHeight+i+1)
-			}
 
-			if err := bcp.historyDB.ProcessBlock(b); err != nil {
-				return err
-			}
+		if err := bcp.historyDB.ProcessBlock(b); err != nil {
+			return err
 		}
-		// logger.Critical("historydb parse %d/%d", bcHeight, bcHeight)
 	}
-	bcp.parsedHeight = bcHeight
-	return nil
+
+	if parsedHeight < int64(bcHeight) {
+		logger.Info("parse block from %d to %d", parsedHeight+1, bcHeight)
+	}
+
+	return bcp.historyDB.SetParsedHeight(bcHeight)
```

Comment:
is it possible for `bcHeight < parsedHeight`? if so i'm not sure it is ok to call `SetParsedHeight` when that occurs. should check for `bcHeight < parsedHeight` as initial condition